fe/fe-core/src/main/java/com/starrocks/qe/QeProcessorImpl.java

```diff
@@ -121,6 +121,8 @@ private void scanMonitorQueries() {
             for (Map.Entry<TUniqueId, Long> entry : monitorQueryMap.entrySet()) {
                 if (now > entry.getValue()) {
                     LOG.warn("monitor expired, query id = {}", DebugUtil.printId(entry.getKey()));
+                    // put profile upload task into profile-worker threadPool, so we can get the most complete profile in FE
+                    getCoordinator(entry.getKey()).finishProfileUpload();
```

Comment:
We could call ` queryProfile.finishAllInstances(Status.OK)` just in `DefaultCoordinator::onFinished`, which is called in `QeProcessorImpl::unregisterQuery`. And then `Coordinator` needn't expose `finishProfileUpload`.

Reply:
fixed